File: sample_app/app/src/main/java/com/tnkfactory/rwd/offerer/MainActivity.java
package com.tnkfactory.rwd.offerer;

import android.content.Context;
import android.content.Intent;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.BaseAdapter;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.tnkfactory.ad.AgreePrivacyPopupListener;
import com.tnkfactory.ad.Logger;
import com.tnkfactory.ad.ServiceCallback;
import com.tnkfactory.ad.TnkSession;
import com.tnkfactory.ad.TnkStyle;
import com.tnkfactory.rwd.offerer.utils.GAIDTask;
import com.tnkfactory.rwd.offerer.viewpager.ViewPagerActivity;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private Boolean isAgreePrivacy = false;   // 개인정보 수집동의 여부

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Logger.enableLogging(true);

        // 유저 식별 값 설정
        new GAIDTask(this, true, new GAIDTask.GAIDListener() {
            @Override
            public void onSuccess(String gaid) {
                if (gaid != null) {
                    TnkSession.setUserName(MainActivity.this, gaid);
                }
            }
        }).execute();

        // 실행형 광고
        TnkSession.applicationStarted(this);

        // COPPA 설정 (true - ON / false - OFF)
        TnkSession.setCOPPA(MainActivity.this, false);

        // Sample Layout 초기화
        initLayout();

        // 오퍼월 애니메이션 설정
        TnkStyle.showAdListAnim_Enter = R.anim.slide_left_from_right;
        TnkStyle.showAdListAnim_Exit = R.anim.zoom_out;
        TnkStyle.finishAdListAnim_Enter = R.anim.zoom_in;
        TnkStyle.finishAdListAnim_Exit = R.anim.slide_right_from_hold;
    }

    @Override
    protected void onResume() {
        super.onResume();

        final TextView pointView = findViewById(R.id.main_point);

        //Tnk 서버에 적립되어 있는 사용자 포인트 값 조회
        TnkSession.queryPoint(this, true, new ServiceCallback() {
            @Override
            public void onReturn(Context context, Object result) {
                Integer point = (Integer)result;
                pointView.setText(String.valueOf(point));
            }
        });
    }

    // Sample layout
    private void initLayout() {
        ListView listView = findViewById(R.id.list_main);
        ArrayList<MainListItem> itemList = new ArrayList<MainListItem>();

        itemList.add(MainListItem.HEADER_01);
        itemList.add(MainListItem.BASIC);
        itemList.add(MainListItem.POPUP_BASIC);
        itemList.add(MainListItem.EMBED_BASIC);
        itemList.add(MainListItem.TEMPLATE);
        itemList.add(MainListItem.TEMPLATE_TAB);

        itemList.add(MainListItem.HEADER_02);
        itemList.add(MainListItem.ViewPager);

        itemList.add(MainListItem.HEADER_03);
        itemList.add(MainListItem.INTERSTITIAL_AD);


        MainListAdapter adapter = new MainListAdapter(this, itemList);
        listView.setAdapter(adapter);

        // List Click Event
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {

                Intent intent = null;

                MainListItem item = (MainListItem) view.getTag();
                switch (item) {
                    case BASIC:
                        TnkSession.showAdList(MainActivity.this, "Basic");
                        break;
                    case POPUP_BASIC:
                        TnkSession.popupAdList(MainActivity.this,  "Popup");
                        break;
                    case EMBED_BASIC:
                        intent = new Intent(MainActivity.this, OfferwallEmbedActivity.class);
                        break;
                    case TEMPLATE:
                        intent = new Intent(MainActivity.this, OfferwallTemplateActivity.class);
                        break;
                    case TEMPLATE_TAB:
                        intent = new Intent(MainActivity.this, TabOfferwallTemplateActivity.class);
                        break;


                    case ViewPager:
                        // 최초 1회는 TnkSession.showAgreePrivacyPopup()를 사용하여 개인정보 수집동의를 받고
                        // 오퍼월을 사용해야 오퍼월 진입시 개인정보 수집동의 팝업이 뜨지 않습니다.
                        // 이 과경을 생략할 경우 뷰페이저로 오퍼월 2개 사용시 개인정보 수집동의 팝업이 2번 노출됩니다.
                        if (isAgreePrivacy) {
                            intent = new Intent(MainActivity.this, ViewPagerActivity.class);
                        } else {
                            TnkSession.showAgreePrivacyPopup(MainActivity.this, new AgreePrivacyPopupListener() {
                                @Override
                                public void onConfirm() {
                                    Toast.makeText(MainActivity.this, "개인정보 수집 동의", Toast.LENGTH_LONG).show();

                                    Intent intent = new Intent(MainActivity.this, ViewPagerActivity.class);
                                    startActivity(intent);
                                    isAgreePrivacy = true;
                                }

                                @Override
                                public void onCancle() {
                                    Toast.makeText(MainActivity.this, "개인정보 수집 미동의", Toast.LENGTH_LONG).show();
                                }
                            });
                        }
                        break;

                    case INTERSTITIAL_AD:
                        TnkSession.prepareInterstitialAd(MainActivity.this, TnkSession.PPI);
                        TnkSession.showInterstitialAd(MainActivity.this);
                        break;
                }

                if (intent != null) {
                    startActivity(intent);
                }
            }
        });
    }

    // Sample list adapter
    public class MainListAdapter extends BaseAdapter {

        private Context context;
        private ArrayList<MainListItem> list;

        public MainListAdapter(Context context, ArrayList<MainListItem> list) {
            this.context = context;
            this.list = list;
        }

        @Override
        public int getCount() {
            if (list != null) {
                return list.size();
            }
            return 0;
        }

        @Override
        public Object getItem(int position) {
            return list.get(position);
        }

        @Override
        public long getItemId(int position) {
            return 0;
        }

        @Override
        public View getView(int position, View view, ViewGroup viewGroup) {
            MainListItem data = list.get(position);
            if (view == null) {
                LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
                view = inflater.inflate(data.isHeader() == true ? R.layout.main_list_header : R.layout.main_list_item, null);
            }

            TextView txtItem = view.findViewById(R.id.text);
            txtItem.setText(data.getValue());
            view.setTag(data);

            return view;
        }
    }
}

File: sample_app_kotlin/app/src/main/java/com/tnkfactory/rwd/offerer/MainActivity.kt
package com.tnkfactory.rwd.offerer

import android.os.Bundle
import android.view.View.OnClickListener
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.tnkfactory.ad.TnkSession
import com.tnkfactory.rwd.offerer.databinding.ActivityMainBinding
import com.tnkfactory.rwd.offerer.databinding.RowMenuItemBinding
import com.tnkfactory.rwd.offerer.utils.GAIDTask
import com.tnkfactory.rwd.offerer.utils.TnkProgressDialog
import com.xwray.groupie.GroupieAdapter
import com.xwray.groupie.databinding.BindableItem
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        // TNK SDK 기본 설정
        initTnkSdk()


        menuItem(binding.rvMainContent)


    }


    private fun initTnkSdk() {

        val progress = TnkProgressDialog(this@MainActivity)
        progress.show()

        lifecycleScope.launch(Dispatchers.IO) {

            // Analytics Report 사용을 위한 설정
            TnkSession.applicationStarted(this@MainActivity)

            // 유저 식별 값 설정
            // 샘플에서는 GAID를 사용해 유저 식별 값을 설정했으나 GAID는 초기화가 가능한 값이므로
            // 사용자가 변경 불가능한 값을 사용해 설정하는 것을 추천합니다.
            val gaid = GAIDTask().getGAID(this@MainActivity)
            TnkSession.setUserName(this@MainActivity, gaid)

            // COPPA 설정 (true - ON / false - OFF)
            TnkSession.setCOPPA(this@MainActivity, false)

        }
        progress.dismiss()
    }


    fun menuItem(recyclerView: RecyclerView) {

        val rvAdapter = GroupieAdapter().apply {
            add(MenuItem("basic") { BasicSampleActivity.start(this@MainActivity) })
            add(MenuItem("embed") { EmbedSampleActivity.start(this@MainActivity) })
            add(MenuItem("custom") { CustomSampleActivity.start(this@MainActivity) })
        }

        recyclerView.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            adapter = rvAdapter
        }


    }


    class MenuItem(private val title: String, private val onClick: OnClickListener) : BindableItem<RowMenuItemBinding>() {
        override fun bind(viewBinding: RowMenuItemBinding, position: Int) {
            viewBinding.tvMenuTitle.text = title
            viewBinding.tvMenuTitle.setOnClickListener(onClick)
        }

        override fun getLayout(): Int = R.layout.row_menu_item

    }
}